### PR TITLE
Encrypt and decrypt content

### PR DESCRIPTION
**Pull request by javiesses**
- Added `encryption-manager` to manage encryption
- Updated README
- Updated existing tests
- Implement encryption in `create` and `swap` methods
- Implemente decryption in `get` method

### CODE REVIEW

File: modules/ipfs-cpinner-client/README.md [hunk 1]
```diff
@@ -33,17 +35,21 @@ const storage: ClientKeyValueStorage = myCustomStorage
 const serviceDid = 'did:ethr:rsk:0x123456789....abc'
 const address = '0xabcdef....123' // user's address
 const did = `did:ethr:rsk:${address}`
-const rpcPersonalSign = (data: string) => window.ethereum.request({ method: 'personal_sign', params: [address, data] }) // this is an example with Metamask
 
-const client = new DataVaultWebClient({ serviceUrl, did, rpcPersonalSign, serviceDid, storage })
+// these are examples with Metamask
+const rpcPersonalSign = (data: string) => window.ethereum.request({ method: 'personal_sign', params: [address, data] })
+const decrypt = (hexCypher: string) => window.ethereum.request({ method: 'eth_decrypt', params: [hexCypher, address] })
+const getEncryptionPublicKey = () => window.ethereum.request.request({ method: 'eth_getEncryptionPublicKey', params: [address] })
```
Comment (ilanolkies) [inline modules/ipfs-cpinner-client/README.md:42]: We can get it once. It should be always the same for the same address

Comment (ilanolkies) [reply modules/ipfs-cpinner-client/README.md:42]: https://www.freecodecamp.org/news/understanding-memoize-in-javascript-51d07d19430e/

**Comment by javiesses** [comment]
Closes #14

File: modules/ipfs-cpinner-client/src/index.ts [hunk 2]
```diff
@@ -19,16 +20,22 @@ const ClientKeyValueStorageFactory = {
...
     this.storage = config.storage || ClientKeyValueStorageFactory.fromLocalStorage()
     this.authManager = authManagerFactory(config, this.storage)
+    this.encryptionManager = encryptionManager(config.getEncryptionPublicKey, config.decrypt)
   }
 
-  get ({ did, key }: GetContentPayload): Promise<GetContentResponsePayload[]> {
-    return axios.get(`${this.config.serviceUrl}/content/${did}/${key}`)
-      .then(res => res.status === 200 && res.data)
-      .catch(this.errorHandler)
+  async get ({ did, key }: GetContentPayload): Promise<GetContentResponsePayload[]> {
```
Comment (ilanolkies) [inline modules/ipfs-cpinner-client/src/index.ts:31]: Discussion: should this method encrypt the `key` or be authenticated?

**Review by ilanolkies — APPROVED** [review]
PR approved, but opened two new issues: #42 and #43

**Pull request merged**